salt/returners/postgres.py

```diff
@@ -269,6 +272,25 @@ def event_return(events):
             cur.execute(sql, (tag, salt.utils.json.dumps(data), __opts__["id"]))
 
 
+def all_minions(tgt,tgt_type):
+    """
+    Retrive all the minions which are expected to return for a job based of the target or target type passed.
+    """
+    if tgt_type == 'glob':
+        salt_command = ['salt',tgt,'--preview-target','--out=text']
+    elif tgt_type == 'compound' or tgt_type == 'list' or tgt_type == 'grain':
+        if tgt_type == "compound":
+            tgt_type = '-C'
+        elif tgt_type == "list":
+            return tgt
+        elif tgt_type == "grain":
+            tgt_type = '-G'
+        salt_command=['salt',tgt_type,tgt,'--preview-target','--out=text']
+    result = subprocess.run(salt_command, capture_output=True, text=True, check=True)
```

Comment:
you dont want to do this; use the minion data cache instead. ill be upstreaming a sqlalchemy returner soon that handles all of this